### PR TITLE
Use `MaybeUninit<u8>` instead of `mem::uninitialized`.

### LINKED ISSUE #35: Use MaybeUninit instead of mem::uninitialized() or somehow remove unitialized memory usage
mem::uninitialized() was used to put outgoing messages on stack and avoid hitting allocator for each and every message. The same effect could be achieved in safe code by re-using the same allocated storage.

Care must be taken to ensure that logged messages do not exceed android allowed message length. It would also be great if the current behavior of splitting too long message at new lines is preserved.

### PR DESCRIPTION
**Pull request by Nilstrieb**
Fixes #35. Also contains a small stacked borrows soundness fix for
a `ptr::copy` invocation, it reorders the creation of the src and dst
pointers.

The code now passes miri with `-Zmiri-tag-raw-pointers`.

The issue did mention waiting for `maybe_uninit_uninit_array`, but this is not required, I have added a little helper that does the same.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -338,21 +338,22 @@ impl<'a> PlatformLogWriter<'a> {
...
+        unsafe { *self.buffer.get_unchecked_mut(len) = last_byte };
     }
 
     /// Copy `len` bytes from `index` position to starting position.
     fn copy_bytes_to_start(&mut self, index: usize, len: usize) {
-        let src = unsafe { self.buffer.as_ptr().add(index) };
         let dst = self.buffer.as_mut_ptr();
+        let src = unsafe { self.buffer.as_ptr().add(index) };
         unsafe { ptr::copy(src, dst, len) };
     }
```
Comment (badboy) [inline src/lib.rs:358]: This whole function block should probably simply be within an unsafe block and document the assumption that make it safe.
Or better make the function unsafe (`unsafe fn`) and still document the requirements. Callers would have to ensure it's correct (even if that's all internally). Maybe even have `debug_assert!`s to ensure index and len are within bounds?

**Review by badboy — APPROVED** [review]
unsolicited review (because I depend on this crate):
this change looks good to me mostly.
I'd like to see this get merged and released. Any chance to make that happen?

**Review by Dushistov — APPROVED** [review]
I tested it on real device, all looks fine.

**Comment by Nercury** [comment]
@badboy @Dushistov @Nilstrieb this is released, sorry for the big delay

**Pull request merged**